### PR TITLE
RavenDB-18645 - Sharding - Studio - Move handlers to processors & imp…

### PR DESCRIPTION
**Pull request by shiranshalom**
…lement missing - Handle NotModified

### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-18645/Sharding-Studio-Move-handlers-to-processors-implement-missing

### Type of change

- Task

### How risky is the change?

- Low 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- No documentation update is needed 

### Testing 

- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

File: src/Raven.Server/Web/Studio/Processors/AbstractStudioCollectionsHandlerProcessorForPreviewCollection.cs [hunk 1]
```diff
@@ -58,7 +58,7 @@ protected virtual ValueTask InitializeAsync()
 
     protected abstract ValueTask<long> GetTotalResultsAsync();
 
-    protected abstract bool NotModified(out string etag);
+    protected abstract ValueTask<bool> NotModified();
```
Comment (ppekrol) [inline src/Raven.Server/Web/Studio/Processors/AbstractStudioCollectionsHandlerProcessorForPreviewCollection.cs]: NotModifiedAsync :)

Comment (shiranshalom) [reply src/Raven.Server/Web/Studio/Processors/AbstractStudioCollectionsHandlerProcessorForPreviewCollection.cs]: Done :)

File: src/Raven.Server/Web/Studio/Sharding/Processors/ShardedStudioCollectionsHandlerProcessorForPreviewCollection.cs [hunk 2]
```diff
@@ -84,12 +84,41 @@ protected override async ValueTask<long> GetTotalResultsAsync()
...
+        }
+        else
+        {
+            var result = await _requestHandler.ShardExecutor.ExecuteParallelForAllAsync(new ShardedLastChangeVectorForCollectionOperation(RequestHandler, Collection, RequestHandler.DatabaseName));
+            foreach ((_, string changeVector) in result.LastChangeVectors)
+            {
+                if (changeVector != null)
+                    changeVectors.Add(changeVector);
+            }
+        }
```
Comment (karmeli87) [inline src/Raven.Server/Web/Studio/Sharding/Processors/ShardedStudioCollectionsHandlerProcessorForPreviewCollection.cs]: why not simply do `ShardedStudioCollectionsHandlerProcessorForPreviewCollection` for each shard?

**Review by karmeli87 — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Web/Studio/Sharding/Processors/ShardedStudioCollectionsHandlerProcessorForPreviewCollection.cs (hunk 2 above)
Comment (karmeli87) [reply src/Raven.Server/Web/Studio/Sharding/Processors/ShardedStudioCollectionsHandlerProcessorForPreviewCollection.cs]: I don't think you need to do any additional requests in `NotModified` ... you should be able to get it from the `ShardedCollectionPreviewOperation` request to each of the shard

Comment (shiranshalom) [reply src/Raven.Server/Web/Studio/Sharding/Processors/ShardedStudioCollectionsHandlerProcessorForPreviewCollection.cs]: True. Done :)

File: src/Raven.Server/Documents/ComputeHttpEtags.cs [hunk 3]
```diff
@@ -119,6 +119,27 @@ public static unsafe string ComputeEtagForRevisions(List<Document> revisions)
             return FinalizeHash(size, state);
         }
 
+        public static unsafe string ComputeEtagForChangeVectors(List<string> changeVectors)
```
Comment (ppekrol) [inline src/Raven.Server/Documents/ComputeHttpEtags.cs]: where is this used?

Comment (shiranshalom) [reply src/Raven.Server/Documents/ComputeHttpEtags.cs]: Nowhere, done

File: src/Raven.Server/Web/Studio/Sharding/Processors/ShardedStudioCollectionsHandlerProcessorForPreviewCollection.cs [hunk 4]
```diff
@@ -84,12 +83,16 @@ protected override async ValueTask<long> GetTotalResultsAsync()
...
 
-    protected override bool NotModified(out string etag)
+    protected override async ValueTask<bool> NotModifiedAsync()
     {
-        DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal,
-            "RavenDB-19066 Need to figure out the best way to combine ETags and send not modified. See `null` passed as etag to above new ShardedCollectionPreviewOperation()");
+        var etag = RequestHandler.GetStringFromHeaders(Constants.Headers.IfNoneMatch);
 
-        etag = null;
+        var result = await _requestHandler.ShardExecutor.ExecuteParallelForAllAsync(new ShardedCollectionPreviewOperation(RequestHandler, etag, _continuationToken));
```
Comment (karmeli87) [inline src/Raven.Server/Web/Studio/Sharding/Processors/ShardedStudioCollectionsHandlerProcessorForPreviewCollection.cs:90]: If I get it right you will do a request to all shards, just to get the not modified and then you will do the same request again in the `GetDocumentsAsync` ?

This brings me to a conclusion that `NotModifiedAsync` shouldn't be async at all..

**Review by karmeli87 — CHANGES_REQUESTED** [review]

**Comment by shiranshalom** [comment]
Moved to https://github.com/ravendb/ravendb/pull/15222

**Pull request closed without merging**